mev-boost-rs/src/relay_mux.rs

```diff
@@ -124,18 +126,40 @@ impl BlindedBlockProvider for RelayMux {
         &self,
         bid_request: &BidRequest,
     ) -> Result<SignedBuilderBid, BlindedBlockProviderError> {
-        let responses = stream::iter(self.relays.iter().cloned())
-            .map(|relay| async move { relay.fetch_best_bid(bid_request).await })
+        let try_responses = stream::iter(self.relays.iter().cloned())
+            .enumerate()
+            .map(|(relay_index, relay)| async move {
+                let bid = tokio::time::timeout(
+                    std::time::Duration::from_secs(FETCH_BEST_BID_TIME_OUT as u64),
+                    relay.fetch_best_bid(bid_request),
+                )
+                .await;
+                (relay_index, bid)
+            })
             .buffer_unordered(self.relays.len())
             .collect::<Vec<_>>()
             .await;
 
+        let (responses, timed_out): (Vec<_>, Vec<_>) =
+            try_responses.into_iter().partition(|(_, res)| res.is_ok());
```

Comment:
we may want to partition here on the error matching `tokio::time::error::Elapsed`, as a relay failure here for some other reason would be counted as a time out

Reply:
see below comment, I'd actually just skip the use of `partition`